web3.js/src/vote-account.ts

```diff
@@ -165,7 +165,9 @@ export class VoteAccount {
       votes: va.votes,
       rootSlot,
       authorizedVoters: va.authorizedVoters.map(parseAuthorizedVoter),
-      priorVoters: va.priorVoters.isEmpty ? [] : va.priorVoters.buf.map(parsePriorVoters),
+      priorVoters: va.priorVoters.isEmpty
+        ? []
+        : va.priorVoters.buf.slice(va.priorVoters.idx).map(parsePriorVoters),
```

Comment:
you also have to append the slice before `idx` to the end

Reply:
oh, to make sure I understand, you mean something along the lines of;

```js
const buf = va.priorVoters.buf;
const arrayToAppend = buf.slice(0, idx - 1);
const parsedVoters = buf.slice(idx).map(parsePriorVoters);
const priorVoters = [...parsedVoters, ...arrayToAppend];
```

Reply:
Yes, exactly!